uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Form/CreateUserRecordTaskForm.php

```diff
@@ -174,4 +176,34 @@ protected function getContactsWithout($property) {
     return array_filter($this->contactDetails, $checker);
   }
 
+  /**
+   * Returns contacts with work emails that are already in use and pairs of new
+   * Contacts that have duplicate emails
+   *
+   * @return array
+   */
+  private function getEmailConflictContacts() {
+    $newAccounts = $this->getContactsWithout('uf_id');
+    $haveNoEmail = $this->getContactsWithout('email');
+    $newContactsWithEmail = array_diff_key($newAccounts, $haveNoEmail);
+
+    $newEmails = array_column($newContactsWithEmail, 'email');
+    $params = ['uf_name' => ['IN' => $newEmails], 'options' => ['limit' => 0]];
+    $existing = civicrm_api3('UFMatch', 'get', $params);
+    $existing = ArrayHelper::value('values', $existing, []);
+    $existingEmails = array_column($existing, 'uf_name');
```

Comment:
The contacts without emails are displayed separately in the summary page.

This function just gets selected contacts that have an email _and_ no drupal account. These are the contacts that we would normally create an account for because they meet all the requirements. 

But first we need to check if any of the contacts are using an existing email (belonging to a Drupal account) as their primary email, if they are then we exclude them from the accounts to be created.